src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -498,6 +500,40 @@ private static void BindProperty(PropertyInfo property, object instance, IConfig
             }
         }
 
+        [RequiresUnreferencedCode("Cannot statically analyze what the element type is of the object collection so its members may be trimmed.")]
+        private static void BindExistingCollection(IEnumerable source, IConfiguration config, BinderOptions options)
+        {
+            IConfigurationSection[] children = config.GetChildren().ToArray();
+            Type sourceType = source.GetType();
+
+            Type? elementType = sourceType.GenericTypeArguments[0];
+            var addMethod = sourceType.GetMethod("Add");
+
+            var singleItemArray = new object[1];
+
+            for (int i = 0; i < children.Length; i++)
+            {
+                try
+                {
+                    object? item = BindInstance(
+                        type: elementType,
+                        instance: null,
+                        config: children[i],
+                        options: options);
+                    if (item != null)
+                    {
+                        singleItemArray[0] = item;
+                        addMethod!.Invoke(source, singleItemArray);
```

Comment:
We don't want to mutate the source. Please create and return a new list containing source elements and the children elements.

Refer to how similarly with `BindArray` we create a new array and return that one.